rundeckapp/grails-app/services/rundeck/services/NotificationService.groovy

```diff
@@ -232,6 +265,37 @@ public class NotificationService implements ApplicationContextAware{
                         destarr = destrecipients.split(', *') as List
                     }
 
+                    def isFormatted =false
+                    if( grailsApplication.config.rundeck.mail."${source.project}"."${source.jobName}".template.log.formatted in [true,'true']){
+                        isFormatted=true
+                    }else if( grailsApplication.config.rundeck.mail."${trigger}".template.log.formatted in [true,'true']){
+                        isFormatted=true
+                    }else if( grailsApplication.config.rundeck.mail.template.log.formatted in [true,'true']){
+                        isFormatted = true
+                    }
+                    StringBuffer outputBuffer = null
+                    def attachlogbody = false
+                    def attachlog=false
+                    if (trigger != 'start' && configAttachLog in ['true',true]) {
+                        if (configAttachLogInline in ['true',true]) {
+                            attachlogbody = true
+                        }
+                        if (configAttachLogInFile in ['true',true]) {
+                            attachlog = true
+                        }
+                    }
+
+                    if(attachlogbody){
+                        outputBuffer=copyExecOutputToStringBuffer(exec,isFormatted)
```

Comment:
if the template doesn't contain `${logoutput.data}` maybe we can skip this, otherwise it is loading all the output into memory, it may be a lot of data.